lxd/device/gpu_physical.go

```diff
@@ -75,6 +75,12 @@ func (d *gpuPhysical) validateConfig(instConf instance.ConfigReader) error {
 				return fmt.Errorf(`Cannot use %q when "id" is set`, field)
 			}
 		}
+
+		// If user requests CDI in conjunction with nvidia.runtime=true we should forbid that.
+		_, err := cdi.ToCDI(d.config["id"])
+		if err == nil && shared.IsTrue(instConf.ExpandedConfig()["nvidia.runtime"]) {
```

Comment:
we could make this a bit more efficient by testing `shared.IsTrue(instConf.ExpandedConfig()["nvidia.runtime"]) ` first and then checking `cdi.ToCDI` (as that does more parsing).

Reply:
sure, I'll rework that one. Thanks!

Reply:
Yes I agree with @tomponline :)

Reply:
>We should also check if if d.config["id"] != "" and not check either if id not specified, as that indicates CDI mode cannot be in use.

We are inside `if d.config["id"] != ""` block already ;-)